hips/tiles/tile.py

```diff
@@ -189,15 +189,13 @@ def _from_raw_data(cls, meta: HipsTileMeta, raw_data: BytesIO) -> 'HipsTile':
                     data = hdu_list[0].data
                     header = hdu_list[0].header
             return cls(meta, data, header)
-        elif meta.file_format == 'jpg':
+        elif meta.file_format in {'jpg', 'png'}:
             with Image.open(raw_data) as image:
                 data = np.array(image)
             return cls(meta, data)
-        elif meta.file_format == 'png':
-            raise NotImplementedError()
         else:
             raise ValueError(f'Tile file format not supported: {meta.file_format}. '
-                             'Supported formats: fits, jpg, png')
+                              'Supported formats: fits, jpg, png')
```

Comment:
I think there's a bug here (that I introduced earlier, not you here): `meta.file_format` doesn't exist, you'd have to use either `self.meta.file_format` or better the local variable `file_format` defined above.

This is pretty common that error handling code (like in this case an invalid `file_format` isn't tested, and then the error handling code has bugs and users get super confused if they hit that code and get a chained exception.
So bottom line: even if it's annoying / quite some work, I would suggest we try to get 100% or almost 100% test coverage and add tests exercising such lines as this one.

The way to do it is to prepare a test case that will reach this line of code and then to call this method inside a `pytest.raises(ValueError)` block, which makes sure the expected error gets raises (and not a NameError like with the current buggy code):
https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions